### ISSUE TITLE
osk (keyboard) dismissing is not intuitive

### ISSUE THREAD

**Issue by userj** [ubports/ubports-touch#77]
**Device: bacon
Channel: legacy
Version: Ubuntu 15.04 (r24)**

**What happens:**
OSK appears when typing in a text field. Sometimes the OSK covers part of the screen and user wants to dismiss it. Swiping down from the top of the OSK makes it go away, but there is no indicator or markings showing that is how to dismiss it.

**Expected:**
A down arrow or some symbol near the top of the keyboard to indicate that it can be swiped away.

**Steps to reproduce:**
-Open Browser App
-Visit website with text field input: eg. github.com
-Tap the search field, OSK appears

**Comment by NeoTheThird** [comment]
Maybe a section about it can be added to the setting up tutorial (possibly when you open the keyboard for the first time), but imho swiping down feels pretty intuitive to me.

**Comment by userj** [comment]
@NeoTheThird Yes, I think it's intuitive for those familiar with such things. On recent review, I find that the problem was more with the way that websites were displayed than the keyboard. It seems to be less intrusive now that ubports displays the mobile versions of sites.

I think it's the fact that I feel the need to dismiss the keyboard at all that is a problem, but not how to do it necessarily. Something about the workflow felt weird.

**Comment by mateosalta** [comment]
Perhaps a way to keep the keyboard hidden when using heavy text fields, that don't nessaraly need typing example might be a spread sheet where you want to click on and navigate, but not allways. Kind of like in the terminal app, but would only hide on a button press then show an overlay button?

**Comment by NeoTheThird** [comment]
Removed from milestone, this needs some more discussion.

**Comment by userj** [comment]
I still feel like some kind of arrow or indicator on the keyboard would be worth discussing. 

Maybe something to show that it has an 'active' edge. Or the first idea of putting it in the tutorial.

**Comment by Arc676** [comment]
Personally, I agree with userj. Originally, the only way I could dismiss the keyboard was by pulling down the system tray to check my battery (or any other icon). Only when I tried to multitask and accidentally dragged downwards did I discover that dragging down on the keyboard would dismiss it (OTA-15 flo).

AFAIAM, other devices do not have this feature.

**Comment by ChloeWolfieGirl** [comment]
You don't have to pull the keyboard down from the top, you can pull it down from the space bar, doens't matter where.

I'd say just have it on the tutorial, or bounce a little.

There's indication that you have to pull it down, as it slides up when you tap on a text box.
Its subtle suggestion, which is usually better then anything big.

**Comment by Danfro** [comment]
Buggrit! I am using my BQ E5 for a couple of years now.
@PhoenixLandPirate: Thanks for that info!
By reading this thread from last year I did learn that swiping down the osk works pretty much everywhere on the keyboard. I was always fiddling around to hit the top edge for swiping down.

Tutorial: definitely! 
I do agree, that a visible edge would make it more intuitive for newbies to swipe the osk away.
But an visible edge does only make sense - in my opinion - if this edge is the only place for swiping. People will tend to tap/swipe only at this edge. And if you do not hit it exactly you will end up - as myself - to think that it does not work or is tricky. If you can swipe everywhere on the osk a visible edge is not necessary if you do get that info from a tutorial. Just make sure people know about that function. ;-)

**Comment by dobey** [comment]
Swiping being the only means of dismissal is neither accessible, intuitive, nor convergent. We need to cover all of these cases to have a superior product. There needs to be a way to dismiss with a tap or mouse click, so that people whom are unable to make the swiping gesture can close the OSK, and this can also include a small visual hint to make it more intuitive. This will also help make the OSK more convergent.

**Comment by mateosalta** [comment]
How about a dismiss on a certain amount of scrolling - ex. User scrolls away from field, is logical they wish to view the page instead of imputing data, reintroduce osk on taping input again, but have a threashold of a certain scroll distance

**Comment by ruedigerkupper** [comment]
I honestly had never discovered the swipe gesture until someone mentionned it here. "People who are familiar with souch things will discover it" is not an argument. Usage patterns need to either readily disclose themselves or be obviously discoverable. This was not, at least not for me.

**Comment by mateosalta** [comment]
true, our default visual indication of being able to swipe is the grey three dot bar - which the keyboard doesn't have - we set up a visual language for this ability and are not using it
